compiler/modules/front/codeGen.bal

```diff
@@ -766,10 +766,19 @@ function codeGenVarDeclStmt(CodeGenContext cx, bir:BasicBlock startBlock, Enviro
     if lookup(varName, env) !== () {
         return cx.semanticErr(`duplicate declaration of ${varName}`);
     }
-    t:SemType semType = check cx.resolveTypeDesc(td);
-    bir:Register result = cx.createRegister(semType, varName);
-    bir:BasicBlock nextBlock = check codeGenAssign(cx, env, startBlock, result, initExpr, semType);
-    return { block: nextBlock, bindings: { name: varName, reg: result, prev: env.bindings, isFinal } };  
+    bir:Register result;
+    bir:BasicBlock nextBlock;
+    if varName == "_" {
+        result = cx.createRegister(t:ANY, varName);
+        nextBlock = check codeGenAssign(cx, env, startBlock, result, initExpr, t:ANY);
+        return { block: nextBlock };
+    }
```

Comment:
I have followed this approach because rhs should belong to type any if lhs has wild-card-binding-pattern. Therefore `error _ = error("Error");` should be an error.
@jclark, WDYT about this approach?